crates/bevy_dev_tools/src/fps_overlay.rs

```diff
@@ -0,0 +1,164 @@
+use bevy_app::{Plugin, Startup, Update};
+use bevy_asset::AssetServer;
+use bevy_color::Color;
+use bevy_diagnostic::{DiagnosticsStore, FrameTimeDiagnosticsPlugin};
+use bevy_ecs::{
+    change_detection::DetectChanges,
+    component::Component,
+    query::With,
+    system::{Commands, Query, Res, Resource},
+};
+use bevy_input::{keyboard::KeyCode, ButtonInput};
+use bevy_render::view::Visibility;
+use bevy_text::{Text, TextSection, TextStyle};
+use bevy_ui::node_bundles::TextBundle;
+
+#[derive(Default)]
+/// A plugin that adds an FPS overlay to the Bevy application.
+pub struct FpsOverlayPlugin {
+    /// Starting configuration of overlay, this can be later be changed through `[FpsOverlayConfig]` resource.
+    pub config: FpsOverlayConfig,
+}
+
+impl Plugin for FpsOverlayPlugin {
+    fn build(&self, app: &mut bevy_app::App) {
+        if !app.is_plugin_added::<FrameTimeDiagnosticsPlugin>() {
+            app.add_plugins(FrameTimeDiagnosticsPlugin);
+        }
+        app.insert_resource(self.config.clone())
+            .add_systems(Startup, setup)
+            .add_systems(Update, (customize_text, update_text, toggle_overlay));
+    }
+}
+
+#[derive(Resource, Clone)]
+/// Configuration options for the FPS overlay.
+pub struct FpsOverlayConfig {
+    /// File path for the font file. If set to `None`,the default font is used.
+    /// Note: The overlay won't be visible if you set it to `None` and run without `default_font` feature.
+    pub font_path: Option<String>,
```

Comment:
Any reason not to just accept a `TextStyle` instead of these three individual font properties?

Reply:
The problem is with `font` on `TextStyle`, because if user doesn't provide font this will use default font. I might be missing something, so if font is not an issue, I'm happy to change it

Reply:
Ah, I'm not sure if I fully understand your message, but it is a bit awkward to ask for a `TextStyle`, because the only `Handle<Font>` a user is likely to have access to when initializing the plugin is `Handle<Font>::default()`.

Still, I think accepting a `TextStyle` is probably better because:

- This "passing around an asset path String" for configuring a font or other asset isn't really done anywhere else in the engine.
- Customizing the font in the engine's debug FPS overlay is a bit of a luxury
- Users are able to replace the font that `Handle<Font>::default()` points to with a bit of effort
- They could still change the font later in a system, and if they want to, using a `TextStyle` / font handle might be a bit more convenient.

Reply:
it should be possible to do something like [`app.world.load_asset("path_to_font")`](http://dev-docs.bevyengine.org/bevy/ecs/world/struct.World.html#method.load_asset) during app setup